control/HTTPCacheControl.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * Class HTTPCacheControl
+ *
+ *
+ */
+class HTTPCacheControl {
```

Comment:
I'm not sure about separating `HTTPCacheControl` from the `HTTP` class. Maybe we should just handle the state statically in `HTTP` instead, and then move to an instance-based (and more testable) `HTTP` class in 5.x? It seems weird to have `HTTP::add_cache_headers()` and `HTTP::set_cache_age()` controlling the `Vary`, `ETag` and `Max-Age` aspects of HTTP caching, and then a separate `HTTPCacheControl` class specifically for the `Cache-Control` header. These headers conceptually belong together, right?

Reply:
tl'dr: The Cache-Control header is quite complicated / specialised I felt it needed a dedicated class to manage its state.

The main reason I went with a separate class was just for my sanity in keeping this closely related logic together.

The `Cache-Control` header is a special kind of header with these multiple directives that can be applied, a bit like a header of headers (like Cookies in that respect) and I just felt this was best contained in one single class. Symfony do something semi-similar where their "Header Bag" has a special treatment of the cache control header (https://github.com/symfony/http-foundation/blob/master/HeaderBag.php). 

I don't much like this approach either, but I prefer it to adding it onto `HTTP` and making that more bloated and less focused. Really, I'd have preferred to be managing this on the response object itself, but that seemed a bit too much of a shift.

Reply:
I think Symfony has it right, treating all headers in one API/class, and then specialising with methods like `addCacheControlDirective()`.